components/button/test/button-mixin.html

```diff
@@ -0,0 +1,65 @@
+<!doctype html>
+<html>
+	<head>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, minimum-scale=1, initial-scale=1, user-scalable=yes">
+		<title>ButtonMixin test</title>
+		<script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
+		<script src="/node_modules/mocha/mocha.js"></script>
+		<script src="/node_modules/chai/chai.js"></script>
+		<script src="/node_modules/@polymer/test-fixture/test-fixture.js"></script>
+		<script src="/node_modules/wct-mocha/wct-mocha.js"></script>
+		<script type="module" src="./test-button-elem.js"></script>
+	</head>
+	<body>
+		<test-fixture id="basic">
+			<template>
+				<test-button-elem></test-button-elem>
+			</template>
+		</test-fixture>
+
+		<test-fixture id="disabled">
+			<template>
+				<div>
+					<test-button-elem disabled>Normal Button</test-button-elem>
+				</div>
+
+			</template>
+		</test-fixture>
+
+		<script type="module">
+			describe('ButtonMixin', () => {
+				let button;
+				beforeEach(async() => {
+					button = fixture('basic');
+					await button.updateComplete;
+				});
+
+				it('should use "type" defaulting to "button"', () => {
+					expect(button.type).to.equal('button');
+				});
+
+				it('should call button focus', async() => {
+					return new Promise(resolve => {
+						button.shadowRoot.querySelector('button').addEventListener('focus', resolve, {once: true});
+						button.focus();
+					});
+				});
+
+				it('should stop propagation if button is disabled', async() => {
+					const disabledButton = fixture('disabled');
+					const childButton = disabledButton.querySelector('test-button-elem');
+					await childButton.updateComplete;
+
+					return new Promise((resolve, reject) => {
+						disabledButton.addEventListener('click', () => {
+							reject();
+						}, {once: true});
+						childButton.click();
+						setTimeout(resolve, 2000); // pass test if parent click eventlistener never called
```

Comment:
In practice is 2s needed? Seems like the bubbling should happen pretty quickly.

Reply:
No, it's pretty instantaneous. I'll decrease this.